CONTRIBUTING.md

```diff
@@ -11,7 +11,110 @@ Thank you for your interest in contributing to CUDA Python! Based on the type of
+
+
+## CI infrastructure overview
+
+The CUDA Python project uses a comprehensive CI pipeline that builds, tests, and releases multiple components across different platforms. This section provides a visual overview of our CI infrastructure to help contributors understand the build and release process.
+
+### CI Pipeline Flow
+
+```
+┌─────────────────────────────────────────────────────────────────────────────────────┐
+│                                   TRIGGER EVENTS                                   │
+│  • Push to main branch              • Pull request                                 │
+│  • Push to pull-request/* branches  • Manual workflow dispatch                    │
+└─────────────────────────────────────────────────────────────────────────────────────┘
+                                             │
+                                             ▼
+┌─────────────────────────────────────────────────────────────────────────────────────┐
+│                                    BUILD STAGE                                     │
+│                                                                                     │
+│  ┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐                     │
+│  │   linux-64      │  │ linux-aarch64   │  │     win-64      │                     │
+│  │  Self-hosted    │  │  Self-hosted    │  │ GitHub-hosted   │                     │
+│  │ linux-amd64-cpu8│  │ linux-arm64-cpu8│  │  windows-2022   │                     │
+│  └─────────────────┘  └─────────────────┘  └─────────────────┘                     │
+│                                                                                     │
+│  • Python versions: 3.9, 3.10, 3.11, 3.12, 3.13                                  │
+│  • CUDA version: 13.0.0 (build-time)                                              │
+│  • Components: cuda-core, cuda-bindings, cuda-pathfinder, cuda-python             │
+└─────────────────────────────────────────────────────────────────────────────────────┘
+                                             │
+                                             ▼
+┌─────────────────────────────────────────────────────────────────────────────────────┐
+│                               ARTIFACT STORAGE                                     │
+│                                                                                     │
+│  ┌──────────────────────────┐              ┌──────────────────────────┐            │
+│  │     GitHub Artifacts     │              │      GitHub Cache        │            │
+│  │   • Wheel files (.whl)   │              │ • Build dependencies     │            │
+│  │   • Test artifacts       │              │ • Compiled binaries      │            │
+│  │   • Documentation        │              │ • Python environments    │            │
+│  │   (30-day retention)     │              │ • Package caches         │            │
```

Comment:
The text on GH Cache is inaccurate. We only cache what we called "mini CTK". The rest (i.e. the proxy cache, I assume which was what confused you) is out of our control and it is ok to leave it out.